algorithms/C/tree/avl-tree.c

```diff
@@ -0,0 +1,195 @@
+#include <stdio.h>
+#include <stdlib.h>
+#define max(a,b) (a > b? a: b)
+
+struct node{
+	struct node* left;
+	struct node* right;
+	int height;
+	int val;
+};
+
+struct node* createNode(int val){
+	struct node* n = (struct node*)malloc(sizeof(struct node));
+	n->val = val;
+	n->left = NULL;
+	n->right = NULL;
+	n->height = 0;
+	return n;
+}
+
+int height(struct node* root){
```

Comment:
Do you mean like this?
```C++
const int height(struct node* root)
```
i don't really get what this does

So, i tried something like this
```C++
const int a(){ return 1; }
int main(){
    int b = a();
    std::cout << ++b;
    return 0;
}
```
Here i am storing the return value of a `const` function inside b, it dosen't give me any error and i was able to change the value of b,
what exactly is the purpose of this `const` function if i can change its value by storing it in a `non const` variable?

Reply:
Sorry, after reading my comment again, I see that it doesn't make any sense, please ignore it

you are right, it will not make any big difference here as you are returning by value here.